output/mistral_fluentbit/mistral_fluentbit.c

```diff
@@ -0,0 +1,553 @@
+#include <errno.h>              /* errno */
+#include <fcntl.h>              /* open */
+#include <getopt.h>             /* getopt_long */
+#include <inttypes.h>           /* uint32_t, uint64_t */
+#include <search.h>             /* insque, remque */
+#include <stdbool.h>            /* bool */
+#include <stdio.h>              /* asprintf */
+#include <stdlib.h>             /* calloc, realloc, free */
+#include <string.h>             /* strerror_r */
+#include <sys/stat.h>           /* open, umask */
+#include <sys/types.h>          /* open, umask */
+#include <sys/time.h>           /* gettimeofday */
+
+#include "mistral_plugin.h"
+#include "mistral_fluentbit_tcp.h"
+
+#define VALID_NAME_CHARS "1234567890abcdefghijklmnopqrstvuwxyzABCDEFGHIJKLMNOPQRSTVUWXYZ-_"
+#define MISTRAL_MAX_BUFFER_SIZE 512
+
+static FILE **log_file_ptr = NULL;
+static char *url = NULL;
+static char *auth = NULL;
+
+static struct timeval mistral_plugin_start;
+static struct timeval mistral_plugin_end;
+
+static mistral_fluentbit_tcp_ctx_s fluentbit_tcp_ctx;
+
+static mistral_log *log_list_head = NULL;
+static mistral_log *log_list_tail = NULL;
+
+static char *custom_variables = NULL;
+
+struct saved_resp {
+    size_t size;
+    char *body;
+};
+
+/*
+ * usage
+ *
+ * Output a usage message via mistral_err.
+ *
+ * Parameters:
+ *   name   - A pointer to a string containing arg[0]
+ *
+ * Returns:
+ *   void
+ */
+static void usage(const char *name)
+{
+    /* This may be called before options have been processed so errors will go
+     * to stderr. While this is designed to be run with Mistral to make the
+     * messages understandable on a terminal add an explicit newline to each
+     * line.
+     */
+    mistral_err("Usage:\n"
+                "  %s [-h host] [-p port] [-e file] [-m octal-mode]\n",
+                name);
+    mistral_err("\n"
+                "  --error=file\n"
+                "  -e file\n"
+                "     Specify location for error log. If not specified all errors will\n"
+                "     be output on stderr and handled by Mistral error logging.\n"
+                "\n"
+                "  --host=hostname\n"
+                "  -h hostname\n"
+                "     The hostname of the Elasticsearch server with which to establish a\n"
+                "     connection. If not specified the plug-in will default to \"localhost\".\n"
+                "\n"
+                "  --mode=octal-mode\n"
+                "  -m octal-mode\n"
+                "     Permissions used to create the error log file specified by the -e\n"
+                "     option.\n"
+                "\n"
+                "  --port=number\n"
+                "  -p number\n"
+                "     Specifies the port to connect to on the Elasticsearch server host.\n"
+                "     If not specified the plug-in will default to \"9200\".\n"
```

Comment:
Elasticsearch and 9200 should be replaced with correct values.

Reply:
I have changed this.